src/components/tokenInput/tokenInput.js

```diff
@@ -1,16 +1,42 @@
-import './tokenInput.css';
+import { useState } from "react";
+import "./tokenInput.css";
 
-function TokenInput(props) {
-    const {onSubmit} = props
-    return (
-        <div>
-            <h1>Enter Slack Token</h1>
-            <form onSubmit={onSubmit}>
-                <input type="password" />
-                <input type="submit" />
-            </form>
-        </div>
-    );
+function TokenInput({ login }) {
+  const [token, setToken] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [tokenValid, setTokenValid] = useState(null);
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    setLoading(true);
+    const valid = await login(token);
+    if (!valid) {
+      setTokenValid(false);
+      setLoading(false);
+    }
+  };
+
+  const invalidMessage = <div className="invalid-feedback">invalid token</div>;
+
+  return (
+    <div>
+      <h1>Enter Slack Token</h1>
+      <form onSubmit={handleSubmit}>
+        <label htmlFor="token-input">Slack Token</label>
```

Comment:
Label feels redundant since it says "Enter Slack Token" directly above it 
```suggestion
```

Reply:
label elements should be associated with input elements to improve accessibility as described in the [MDN Documentation for input labels](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#labels).
we can restyle the form elements and/or rename the heading in a future ui revision